pages/listings/[classId].vue

```diff
@@ -66,24 +95,91 @@ const metadataStore = useMetadataStore();
 const { wallet, signer } = storeToRefs(walletStore);
 const listing = ref([] as any[]);
 const writingNFTListing = ref([] as any[]);
+const royaltyConfig = ref(null as any);
 const metadata = ref({} as any);
+const iscnData = ref({} as any);
+const iscnRoyalty = ref(null as any);
 
 const classId = computed(() => route.params.classId as string);
 const combinedListing = computed(() => ([] as any)
   .concat(...listing.value, ...writingNFTListing.value)
   .sort((a: any, b: any) => a.price.toNumber() - b.price.toNumber()));
+const royaltyList = computed(() => {
+  const rateBasisPoints = (royaltyConfig.value?.rateBasisPoints / 100) || 0;
+  const totalWeight = royaltyConfig.value?.stakeholders.reduce((acc: number, s: any) => acc + s.weight, 0) || 1;
+  return royaltyConfig.value?.stakeholders?.map((s: any) => ({ ...s, weight: (s.weight / totalWeight) * rateBasisPoints }))
+})
+const isShowSyncRoyaltyButton = computed(() => {
+  const isReady = !!iscnData.value;
+  const isOwner = iscnData.value.owner === wallet.value;
+  const isNotEqual = iscnRoyalty.value?.length !== royaltyConfig.value?.stakeholders.length;
+  let needsUpdate = false;
+  iscnRoyalty.value?.forEach((data: any, index: number) => {
```

Comment:
How about using `Array.prototype.some` for short circuit evaluation and return true if L119 condition is true